src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs

```diff
@@ -217,12 +217,14 @@ internal static int GetWarningLevel(ErrorCode code)
                 case ErrorCode.WRN_NubExprIsConstBool2:
                 case ErrorCode.WRN_StaticInAsOrIs:
                 case ErrorCode.WRN_PrecedenceInversion:
-                case ErrorCode.WRN_UnassignedThisAutoProperty:
-                case ErrorCode.WRN_UnassignedThis:
+                case ErrorCode.WRN_UseDefViolationPropertyUnsupportedVersion:
+                case ErrorCode.WRN_UseDefViolationFieldUnsupportedVersion:
+                case ErrorCode.WRN_UnassignedThisAutoPropertyUnsupportedVersion:
+                case ErrorCode.WRN_UnassignedThisUnsupportedVersion:
```

Comment:
New warning wave warnings are added. They should go into warning wave 7 (not 5). 
Also, they should be documented in [https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Warnversion%20Warning%20Waves.md](https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Warnversion%20Warning%20Waves.md)

Reply:
I think wave 5 is the right place for these because they are produced under the same conditions that the original variants of these diagnostic IDs are reported. This is strictly to fulfill the objective outlined here: https://github.com/dotnet/csharplang/issues/5737#issuecomment-1071179924

> * If an old compiler gives a definite-assignment **warning** in some scenario, and the new compiler with preview LangVersion does not give that same warning, then the new compiler with old LangVersion should give a LangVersion **warning**.

Reply:
For example, the only place we produce `WRN_UseDefViolationPropertyUnsupportedVersion` is in certain specific scenarios where previously `WRN_UseDefViolationProperty` was produced. It's really the same warning, it's just that in the new compiler we have additional information to give you--specifically, you can make this warning go away by upgrading the language version.

Reply:
Thanks for the clarification. You're right. `WRN_UseDefViolationPropertyUnsupportedVersion` and `WRN_UseDefViolationFieldUnsupportedVersion` should be in the same warning wave as `WRN_UseDefViolationProperty` and `WRN_UseDefViolationField`.